### PR TITLE
feat: [DHIS2-11261] Add new visualization for programs

### PR DESCRIPTION
**Pull request by Sharmyn28**
This PR has:

- Basic components such as Text field to add a title or name a visualization and Select program
- Query to generate a random id
- Radio Group: 
  - new group that shows a text field
  - field to select a group that has been created previously
 

_Dialog to add visualization_
image

_Title field_
image

_Group field_
image

image

image

### CODE REVIEW

File: src/pages/Analytics/Program/helper.js [hunk 1]
```diff
@@ -97,3 +125,43 @@ export const createDataStoreGroupRows = datastore => {
...
+    } else {
+        updatedRow[current.program] = {
+            programName: current.programName,
+            groups: {
+                [current.group.id]: [current],
+            },
+        }
+    }
+
+    return updatedRow
```
Comment (mediremi) [inline src/pages/Analytics/Program/helper.js:166]: I haven't tested it, but the following should be equivalent:

```suggestion
    const programRow = rows[current.program]
    
    if (programRow) {
        const group = programRow.groups[current.group.id]
        const updatedGroups = {
            ...programRow.groups,
            [current.group.id]: group ? [...group, current] : [current]
        }

        return {
            ...rows,
            [current.program]: {
                ...programRow,
                groups: updatedGroups
            }
        }
    }
    
    return {
        ...rows,
        [current.program]: {
            programName: current.programName,
            groups: {
                [current.group.id]: [current],
            }
        }
    }
```

File: src/components/analyticVisualization/SelectProgram.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export const SelectProgram = ({ settings, onChange }) => {
+    const { programList, loading } = useReadProgramQuery()
+    const [options, setOptions] = useState([])
+
+    useEffect(() => {
+        if (programList) {
+            setOptions(programList)
+        }
+    }, [programList])
```
Comment (mediremi) [inline src/components/analyticVisualization/SelectProgram.js:15]: I think this should work too?

```suggestion
    const { programList, loading } = useReadProgramQuery()
    const options = programList || []
```

File: src/pages/Analytics/Program/helper.js [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+    program: initialValues.program || '',
+    visualization: initialValues.visualization || '',
+    name: initialValues.name || '',
+    group: initialValues.group || {
+        name: '',
+        id: '',
+    },
+})
+
+export const validMandatoryFields = settings => {
```
Comment (mediremi) [inline src/pages/Analytics/Program/helper.js:15]: Should this function be called `invalidMandatoryFields` instead? Seems like it's false when `settings.program` and `settings.visualization` are valid

File: src/pages/Analytics/Program/helper.js (hunk 1 above)
Comment (Sharmyn28) [reply src/pages/Analytics/Program/helper.js:166]: Thank you, it's working

**Review by mediremi — APPROVED** [review]

**Pull request merged**